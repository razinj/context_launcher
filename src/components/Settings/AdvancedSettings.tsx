// React
import React from 'react'
// React Native
import { Pressable, PressableAndroidRippleConfig, StyleSheet, ToastAndroid, View } from 'react-native'
// Components
import SettingsItemLabel from './SettingsItemLabel'
// Redux
import { useDispatch } from 'react-redux'
import { setAppsList } from '../../slices/appsList'
import { clearRecentApps } from '../../slices/recentApps'
import { resetPreferences } from '../../slices/preferences'
import { clearFavoriteApps } from '../../slices/favoriteApps'
// Native modules
import AppsModule from '../../native-modules/AppsModule'
// Models
import { AppDetails } from '../../models/app-details'

const settingItemButtonRippleConfig: PressableAndroidRippleConfig = {
  borderless: false,
  foreground: true,
  color: '#e5e5e5',
}

const AdvancedSettings = () => {
  const dispatch = useDispatch()

  const reloadAllApps = () => {
    AppsModule.getApplications((applications: string) => {
      const apps = JSON.parse(applications) as AppDetails[]

      dispatch(setAppsList(apps))
      displayToast('All apps reloaded successfully!')
    })
  }

  const onClearRecentApps = () => {
    dispatch(clearRecentApps())
    displayToast('Recent apps cleared successfully!')
  }

  const onClearFavoriteApps = () => {
    dispatch(clearFavoriteApps())
    displayToast('Favorite apps cleared successfully!')
  }

  const onResetPreferences = () => {
    dispatch(resetPreferences())
    displayToast('Settings reset successfully!')
  }

  const displayToast = (message: string, duration: number = ToastAndroid.LONG) => {
    ToastAndroid.show(message, duration)
  }

  return (
    <>
      {/* Reload all apps */}
      <View style={styles.itemContainer}>
        <Pressable
          testID='advanced-settings-reload-all-apps-button'
          onPress={reloadAllApps}
          android_disableSound={true}
          android_ripple={settingItemButtonRippleConfig}
          style={styles.buttonItemPressable}>
          <SettingsItemLabel title='Reload all apps' />
        </Pressable>
      </View>

      {/* Clear recent apps */}
      <View style={styles.itemContainer}>
        <Pressable
          testID='advanced-settings-clear-recent-apps-button'
          onPress={onClearRecentApps}
          android_disableSound={true}
          android_ripple={settingItemButtonRippleConfig}
          style={styles.buttonItemPressable}>
          <SettingsItemLabel title='Clear recent apps' />
        </Pressable>
      </View>

      {/* Clear favorite apps */}
      <View style={styles.itemContainer}>
        <Pressable
          testID='advanced-settings-clear-favorite-apps-button'
          onPress={onClearFavoriteApps}
          android_disableSound={true}
          android_ripple={settingItemButtonRippleConfig}
          style={styles.buttonItemPressable}>
          <SettingsItemLabel title='Clear favorite apps' />
        </Pressable>
      </View>

      {/* Reset preferences */}
      <View style={styles.itemContainer}>
        <Pressable
          testID='advanced-settings-reset-preferences-button'
          onPress={onResetPreferences}
          android_disableSound={true}
          android_ripple={settingItemButtonRippleConfig}
          style={styles.buttonItemPressable}>
          <SettingsItemLabel title='Reset settings' />
        </Pressable>
      </View>
    </>
  )
}

const styles = StyleSheet.create({
  itemContainer: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    borderBottomWidth: 1,
    borderBottomColor: '#e5e5e5',
  },
  buttonItemPressable: {
    flex: 1,
  },
})

export default AdvancedSettings
