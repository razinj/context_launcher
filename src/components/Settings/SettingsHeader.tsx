import React from 'react'
import { Pressable, PressableAndroidRippleConfig, StyleSheet, Text, View } from 'react-native'
import { useDispatch } from 'react-redux'
import { APP_BUILD_NUMBER, APP_ID, APP_VERSION } from '../../constants'
import { setDisplaySettings } from '../../slices/appState'
import { showAppDetails } from '../../utils/apps-module'
import CustomIcon from '../shared/CustomIcon'

const appInfoIconRippleConfig: PressableAndroidRippleConfig = {
  borderless: true,
  foreground: true,
  color: '#e5e5e5',
  radius: 18,
}

const SettingsHeader = () => {
  const dispatch = useDispatch()

  const onAppInfoClick = () => {
    showAppDetails(APP_ID)
    dispatch(setDisplaySettings(false))
  }

  return (
    <View style={styles.wrapper} testID='settings-header-wrapper'>
      <Text style={styles.title}>
        Context Settings
        <Text style={styles.appInfoText}>
          &nbsp;&nbsp;v{APP_VERSION} ({APP_BUILD_NUMBER})
        </Text>
      </Text>
      <Pressable
        onPress={onAppInfoClick}
        android_disableSound={true}
        android_ripple={appInfoIconRippleConfig}
        testID='settings-header-app-info-button'>
        <CustomIcon name='information-outline' size={34} color='#808080' />
      </Pressable>
    </View>
  )
}

const styles = StyleSheet.create({
  wrapper: {
    paddingBottom: 20,
    alignItems: 'center',
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  title: {
    fontSize: 20,
    color: '#808080',
    fontWeight: '400',
  },
  appInfoText: {
    color: '#ccc',
    fontSize: 12,
  },
})

export default SettingsHeader
