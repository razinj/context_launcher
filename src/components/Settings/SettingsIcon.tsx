// React
import React, { useContext } from 'react'
// React Native
import { Pressable, PressableAndroidRippleConfig, StyleSheet, View } from 'react-native'
// Contexts
import GlobalContext from '../../contexts/GlobalContext'
// Icons
import Icon from 'react-native-vector-icons/MaterialCommunityIcons'
// Constants
import { PRIMARY_HEX_COLOR, SECONDARY_HEX_COLOR } from '../../constants'

const rippleConfig: PressableAndroidRippleConfig = {
  color: SECONDARY_HEX_COLOR,
  borderless: true,
  foreground: true,
  radius: 20,
}

const SettingsIcon = () => {
  const { displaySettingsBottomSheet } = useContext(GlobalContext)

  return (
    <View style={styles.wrapper}>
      <Pressable
        testID='settings-button'
        onPress={displaySettingsBottomSheet}
        android_disableSound={true}
        android_ripple={rippleConfig}>
        <Icon name='dots-vertical' size={34} style={styles.icon} color={PRIMARY_HEX_COLOR} />
      </Pressable>
    </View>
  )
}

const styles = StyleSheet.create({
  wrapper: {
    alignItems: 'center',
    flexDirection: 'row',
  },
  icon: {
    margin: 5,
  },
})

export default SettingsIcon
