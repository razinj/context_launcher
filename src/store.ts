// Redux
import { configureStore, combineReducers } from '@reduxjs/toolkit'
// Reducers
import appsListReducer from './slices/appsList'
import appsSearchReducer from './slices/appsSearch'
import recentAppsReducer from './slices/recentApps'
import FavoriteAppsReducer from './slices/favoriteApps'
import preferencesReducer from './slices/preferences'
// Storage
import AsyncStorage from '@react-native-async-storage/async-storage'
// Redux Persist
import {
  persistStore,
  persistReducer,
  FLUSH,
  REHYDRATE,
  PAUSE,
  PERSIST,
  PURGE,
  REGISTER,
  createMigrate,
} from 'redux-persist'

const rootReducer = combineReducers({
  appsList: appsListReducer,
  appsSearch: appsSearchReducer,
  recentApps: recentAppsReducer,
  favoriteApps: FavoriteAppsReducer,
  preferences: preferencesReducer,
})

const migrations = {
  2: (state: any) => {
    return {
      ...state,
      preferences: {
        displayRecentApps: true,
        displayFavoriteApps: true,
      },
    }
  },
}

const persistConfig = {
  key: 'root',
  version: 2,
  storage: AsyncStorage,
  migrate: createMigrate(migrations, { debug: false }),
}

const persistedReducer = persistReducer(persistConfig, rootReducer)

export const store = configureStore({
  reducer: persistedReducer,
  middleware: getDefaultMiddleware =>
    getDefaultMiddleware({
      serializableCheck: {
        ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER],
      },
    }),
})

export const persistor = persistStore(store)

export type RootState = ReturnType<typeof store.getState>
export type AppDispatch = typeof store.dispatch
