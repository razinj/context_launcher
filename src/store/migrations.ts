import { MigrationManifest } from 'redux-persist'

export const migrations: MigrationManifest = {
  2: (state: any) => {
    return {
      ...state,
      preferences: {
        displayRecentApps: true,
        displayFavoriteApps: true,
      },
    }
  },
  3: (state: any) => {
    return {
      ...state,
      pinnedApps: {
        list: [],
        temporaryAppsConfig: {
          startDate: undefined,
          endDate: undefined,
        },
      },
      preferences: {
        ...state.preferences,
        displayPinnedApps: true,
        displayTemporaryPinnedApps: false,
      },
    }
  },
  4: (state: any) => {
    const localState = { ...state }
    delete localState.appsSearch

    return {
      ...localState,
      appsList: {
        list: [],
      },
      pinnedApps: {
        list: [],
        temporarily: [],
        temporaryAppsConfig: {
          startDate: undefined,
          endDate: undefined,
        },
      },
      appState: {
        displayAllApps: false,
        displaySettings: false,
        menuAppDetails: undefined,
        displayAppMenu: false,
        displaySortableFavoriteApps: false,
        displaySortablePinnedApps: false,
        search: {
          query: undefined,
          result: [],
        },
      },
    }
  },
}
