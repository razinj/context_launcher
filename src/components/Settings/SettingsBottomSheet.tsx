// React
import React, { useContext, useMemo, useState } from 'react'
// React Native
import { Pressable, PressableAndroidRippleConfig, StyleSheet, Switch, Text, View } from 'react-native'
// Components
import AdvancedSettings from './AdvancedSettings'
import SettingsItemLabel from './SettingsItemLabel'
// Redux
import { useDispatch, useSelector } from 'react-redux'
import {
  displayFavoriteApps,
  displayRecentApps,
  selectDisplayFavoriteAppsMemoized,
  selectDisplayRecentAppsMemoized,
} from '../../slices/preferences'
import { selectFavoriteAppsCountMemoized } from '../../slices/favoriteApps'
// Contexts
import GlobalContext from '../../contexts/GlobalContext'
// BottomSheet
import { BottomSheetModal, BottomSheetModalProvider } from '@gorhom/bottom-sheet'
// Icon
import Icon from 'react-native-vector-icons/MaterialCommunityIcons'
// Utils
import { showAppDetails } from '../../utils/appsModule'
// Constants
import {
  PRIMARY_HEX_COLOR,
  CONTEXT_LAUNCHER_APP_ID,
  CONTEXT_LAUNCHER_APP_VERSION,
  CONTEXT_LAUNCHER_APP_BUILD_NUMBER,
} from '../../constants'

const activeSwitch = PRIMARY_HEX_COLOR
const inActiveSwitch = '#f4f3f4'
const switchTrackColor = { false: '#e5e5e5', true: '#e5e5e5' }

const appInfoIconRippleConfig: PressableAndroidRippleConfig = {
  borderless: true,
  foreground: true,
  color: '#e5e5e5',
  radius: 18,
}

const settingItemButtonRippleConfig: PressableAndroidRippleConfig = {
  borderless: false,
  foreground: true,
  color: '#e5e5e5',
}

const SettingsBottomSheet = () => {
  const dispatch = useDispatch()
  const favoriteAppsCount = useSelector(selectFavoriteAppsCountMemoized)
  const displayRecentAppsValue = useSelector(selectDisplayRecentAppsMemoized)
  const displayFavoriteAppsValue = useSelector(selectDisplayFavoriteAppsMemoized)
  const { dismissKeyboard, settingsBottomSheetRef, toggleSortableFavoriteApps } = useContext(GlobalContext)
  const [displayAdvancedSettings, setDisplayAdvancedSettings] = useState(false)

  const toggleDisplayRecentApps = () => {
    dispatch(displayRecentApps(!displayRecentAppsValue))
  }

  const toggleDisplayFavoriteApps = () => {
    dispatch(displayFavoriteApps(!displayFavoriteAppsValue))
  }

  const onFavoriteAppsSortViewClick = () => {
    dismissKeyboard()
    toggleSortableFavoriteApps()
  }

  const onAppInfoClick = () => {
    showAppDetails(CONTEXT_LAUNCHER_APP_ID)
    settingsBottomSheetRef?.current?.close()
  }

  const favoriteAppsSortDisabled = useMemo(
    () => favoriteAppsCount <= 1 || !displayFavoriteAppsValue,
    [favoriteAppsCount, displayFavoriteAppsValue]
  )

  const toggleAdvancedSettings = (value: boolean = !displayAdvancedSettings) => {
    setDisplayAdvancedSettings(value)
  }

  return (
    <BottomSheetModalProvider>
      <BottomSheetModal
        onDismiss={() => toggleAdvancedSettings(false)}
        ref={settingsBottomSheetRef}
        snapPoints={['25%', '50%', '75%', '90%']}
        style={styles.bottomSheetModal}>
        {/* Settings wrapper */}
        <View style={styles.settingsWrapper}>
          {/* Header wrapper */}
          <View style={styles.headerWrapper}>
            <Text style={styles.headerTitle}>
              Context Settings
              <Text style={styles.appInfoText}>
                &nbsp;&nbsp;v{CONTEXT_LAUNCHER_APP_VERSION} ({CONTEXT_LAUNCHER_APP_BUILD_NUMBER})
              </Text>
            </Text>
            <Pressable onPress={onAppInfoClick} android_disableSound={true} android_ripple={appInfoIconRippleConfig}>
              <Icon name='information-outline' size={34} color='#808080' />
            </Pressable>
          </View>

          {/* Settings */}
          {/* Recent apps switch */}
          <View style={styles.itemContainer}>
            <SettingsItemLabel title='Display recent apps' />
            <Switch
              value={displayRecentAppsValue}
              onValueChange={toggleDisplayRecentApps}
              trackColor={switchTrackColor}
              thumbColor={displayRecentAppsValue ? activeSwitch : inActiveSwitch}
            />
          </View>
          {/* Favorite apps switch */}
          <View style={styles.itemContainer}>
            <SettingsItemLabel title='Display favorite apps' />
            <Switch
              value={displayFavoriteAppsValue}
              onValueChange={toggleDisplayFavoriteApps}
              trackColor={switchTrackColor}
              thumbColor={displayFavoriteAppsValue ? activeSwitch : inActiveSwitch}
            />
          </View>
          {/* Sort favorite apps */}
          <View style={styles.itemContainer}>
            <Pressable
              disabled={favoriteAppsSortDisabled}
              onPress={onFavoriteAppsSortViewClick}
              android_disableSound={true}
              android_ripple={settingItemButtonRippleConfig}
              style={[styles.buttonItemPressable, { opacity: favoriteAppsSortDisabled ? 0.5 : 1 }]}>
              <SettingsItemLabel
                title='Sort favorite apps'
                description={
                  favoriteAppsSortDisabled
                    ? displayFavoriteAppsValue
                      ? 'Add more favorite apps to be able to sort'
                      : 'Display favorite apps to be able to sort'
                    : 'Click to start sorting your favorite apps'
                }
              />
            </Pressable>
          </View>
          {/* Advanced settings wrapper */}
          <>
            {/* Toggle advanced settings */}
            <View style={styles.itemContainer}>
              <Pressable
                onPress={() => toggleAdvancedSettings()}
                android_disableSound={true}
                android_ripple={settingItemButtonRippleConfig}
                style={styles.buttonItemPressable}>
                <SettingsItemLabel title='Toggle advanced settings' />
                <Icon name={`chevron-${displayAdvancedSettings ? 'up' : 'down'}`} size={30} color='#808080' />
              </Pressable>
            </View>

            {/* Advanced settings */}
            {displayAdvancedSettings && <AdvancedSettings />}
          </>
        </View>
      </BottomSheetModal>
    </BottomSheetModalProvider>
  )
}

const styles = StyleSheet.create({
  settingsWrapper: {
    paddingHorizontal: 10,
  },
  itemContainer: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    borderBottomWidth: 1,
    borderBottomColor: '#e5e5e5',
  },
  buttonItemPressable: {
    flex: 1,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
  },
  headerWrapper: {
    paddingBottom: 20,
    alignItems: 'center',
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  headerTitle: {
    fontSize: 20,
    color: '#808080',
    fontWeight: '400',
  },
  bottomSheetModal: {
    marginHorizontal: 5,
  },
  appInfoText: {
    color: '#ccc',
    fontSize: 12,
  },
})

export default SettingsBottomSheet
