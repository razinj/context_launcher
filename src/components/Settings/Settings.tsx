import React from 'react'
import { StyleSheet, View } from 'react-native'
import { Modal, Portal } from 'react-native-paper'
import { useDispatch, useSelector } from 'react-redux'
import { BOTTOM_CONTAINER_HEIGHT_WITH_PADDINGS } from '../../constants'
import { selectDisplaySettings, setDisplaySettings } from '../../slices/appState'
import SettingsHeader from './SettingsHeader'
import AdvancedSettings from './sections/AdvancedSettings'
import FavoriteAppsSettings from './sections/FavoriteAppsSettings'
import PinnedAppsSettings from './sections/PinnedAppsSettings'
import RecentAppsSettings from './sections/RecentAppsSettings'
import ToggleSettings from './shared/ToggleSettings'

const Settings = () => {
  const dispatch = useDispatch()
  const displaySettings = useSelector(selectDisplaySettings)

  const closeSettings = () => {
    dispatch(setDisplaySettings(false))
  }

  return (
    <Portal>
      <Modal contentContainerStyle={styles.contentContainerStyle} visible={displaySettings} onDismiss={closeSettings}>
        <View testID='settings-wrapper'>
          <SettingsHeader />

          <ToggleSettings title={'Recent Apps'}>
            <RecentAppsSettings />
          </ToggleSettings>

          <ToggleSettings title={'Pinned Apps'}>
            <PinnedAppsSettings />
          </ToggleSettings>

          <ToggleSettings title={'Favorite Apps'}>
            <FavoriteAppsSettings />
          </ToggleSettings>

          <ToggleSettings title={'Advanced Settings'}>
            <AdvancedSettings />
          </ToggleSettings>
        </View>
      </Modal>
    </Portal>
  )
}

const styles = StyleSheet.create({
  contentContainerStyle: {
    position: 'absolute',
    left: 5,
    right: 5,
    bottom: BOTTOM_CONTAINER_HEIGHT_WITH_PADDINGS,
    padding: 10,
    borderRadius: 5,
    backgroundColor: 'white',
  },
})

export default Settings
