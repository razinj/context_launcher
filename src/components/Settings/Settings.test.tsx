import { screen, within } from '@testing-library/react-native'
import React from 'react'
import { initialStoreState } from '../../../utils/test/data'
import { renderWithProvider } from '../../../utils/test/utils'
import { RootState } from '../../store'
import Settings from './Settings'

describe('<Settings /> Tests', () => {
  it('should not render modal when display settings value is false (default)', () => {
    renderWithProvider(<Settings />)

    expect(screen.queryByTestId('settings-wrapper')).toBeNull()
  })

  it('should render modal correctly and match snapshot when display settings value is false', () => {
    const customInitialState: RootState = {
      ...initialStoreState,
      appState: {
        ...initialStoreState.appState,
        displaySettings: true,
      },
    }

    renderWithProvider(<Settings />, { preloadedState: customInitialState })

    expect(screen.toJSON()).toMatchSnapshot()

    const settingsWrapper = screen.getByTestId('settings-wrapper')

    expect(settingsWrapper).toBeOnTheScreen()
    expect(within(settingsWrapper).getByText('Recent Apps')).toBeOnTheScreen()
    expect(within(settingsWrapper).getByText('Pinned Apps')).toBeOnTheScreen()
    expect(within(settingsWrapper).getByText('Favorite Apps')).toBeOnTheScreen()
    expect(within(settingsWrapper).getByText('Advanced Settings')).toBeOnTheScreen()
  })
})
