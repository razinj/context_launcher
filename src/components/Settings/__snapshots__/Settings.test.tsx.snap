// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Settings /> Tests should render modal correctly and match snapshot when display settings value is false 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    collapsable={false}
    pointerEvents="box-none"
    style={
      {
        "flex": 1,
      }
    }
  />
  <View
    collapsable={false}
    pointerEvents="box-none"
    style={
      {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      accessibilityLiveRegion="polite"
      accessibilityViewIsModal={true}
      collapsable={false}
      onAccessibilityEscape={[Function]}
      pointerEvents="auto"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      testID="modal"
    >
      <View
        accessibilityLabel="Close modal"
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        importantForAccessibility="no"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "backgroundColor": "rgba(50, 47, 55, 0.4)",
            "flex": 1,
            "opacity": 1,
          }
        }
        testID="modal-backdrop"
      />
      <View
        pointerEvents="box-none"
        style={
          [
            {
              "bottom": 0,
              "justifyContent": "center",
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            {
              "marginBottom": 0,
              "marginTop": 0,
            },
            undefined,
          ]
        }
        testID="modal-wrapper"
      >
        <View
          collapsable={false}
          style={
            {
              "backgroundColor": "white",
              "borderRadius": 5,
              "bottom": 58.8,
              "left": 5,
              "opacity": 1,
              "position": "absolute",
              "right": 5,
              "shadowColor": "#000",
              "shadowOffset": {
                "height": 1,
                "width": 0,
              },
              "shadowOpacity": 0.15,
              "shadowRadius": 3,
            }
          }
          testID="modal-surface-outer-layer"
        >
          <View
            collapsable={false}
            style={
              {
                "backgroundColor": "white",
                "borderRadius": 5,
                "flex": undefined,
                "justifyContent": "center",
                "padding": 10,
                "shadowColor": "#000",
                "shadowOffset": {
                  "height": 1,
                  "width": 0,
                },
                "shadowOpacity": 0.3,
                "shadowRadius": 1,
              }
            }
            testID="modal-surface"
          >
            <View
              testID="settings-wrapper"
            >
              <View
                style={
                  {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                    "paddingBottom": 20,
                  }
                }
                testID="wrapper"
              >
                <Text
                  style={
                    {
                      "color": "#808080",
                      "fontSize": 20,
                      "fontWeight": "400",
                    }
                  }
                >
                  Context Settings
                  <Text
                    style={
                      {
                        "color": "#ccc",
                        "fontSize": 12,
                      }
                    }
                  >
                      v
                    1.4.8
                     (
                    12
                    )
                  </Text>
                </Text>
                <View
                  style={
                    {
                      "flexDirection": "row",
                      "gap": 10,
                    }
                  }
                >
                  <View
                    accessibilityState={
                      {
                        "busy": undefined,
                        "checked": undefined,
                        "disabled": undefined,
                        "expanded": undefined,
                        "selected": undefined,
                      }
                    }
                    accessibilityValue={
                      {
                        "max": undefined,
                        "min": undefined,
                        "now": undefined,
                        "text": undefined,
                      }
                    }
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    testID="app-info-button"
                  >
                    <Text
                      allowFontScaling={false}
                      selectable={false}
                      style={
                        [
                          {
                            "color": "#808080",
                            "fontSize": 34,
                          },
                          undefined,
                          {
                            "fontFamily": "Material Design Icons",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          {},
                        ]
                      }
                    >
                      󰋽
                    </Text>
                  </View>
                  <View
                    accessibilityState={
                      {
                        "busy": undefined,
                        "checked": undefined,
                        "disabled": undefined,
                        "expanded": undefined,
                        "selected": undefined,
                      }
                    }
                    accessibilityValue={
                      {
                        "max": undefined,
                        "min": undefined,
                        "now": undefined,
                        "text": undefined,
                      }
                    }
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    testID="system-settings-button"
                  >
                    <Text
                      allowFontScaling={false}
                      selectable={false}
                      style={
                        [
                          {
                            "color": "#808080",
                            "fontSize": 34,
                          },
                          undefined,
                          {
                            "fontFamily": "Material Design Icons",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          {},
                        ]
                      }
                    >
                      󰢻
                    </Text>
                  </View>
                </View>
              </View>
              <View
                style={
                  {
                    "borderRadius": 5,
                  }
                }
              >
                <View
                  accessibilityState={
                    {
                      "busy": undefined,
                      "checked": undefined,
                      "disabled": undefined,
                      "expanded": undefined,
                      "selected": undefined,
                    }
                  }
                  accessibilityValue={
                    {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "space-between",
                    }
                  }
                  testID="toggle-settings-button"
                >
                  <View
                    style={
                      [
                        {
                          "padding": 10,
                        },
                        undefined,
                      ]
                    }
                    testID="wrapper"
                  >
                    <Text
                      style={
                        [
                          {
                            "color": "#808080",
                          },
                          [],
                        ]
                      }
                      testID="title"
                    >
                      Recent Apps
                    </Text>
                  </View>
                  <Text
                    allowFontScaling={false}
                    selectable={false}
                    style={
                      [
                        {
                          "color": "#808080",
                          "fontSize": 30,
                        },
                        undefined,
                        {
                          "fontFamily": "Material Design Icons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        {},
                      ]
                    }
                  >
                    󰅀
                  </Text>
                </View>
              </View>
              <View
                style={
                  {
                    "borderRadius": 5,
                  }
                }
              >
                <View
                  accessibilityState={
                    {
                      "busy": undefined,
                      "checked": undefined,
                      "disabled": undefined,
                      "expanded": undefined,
                      "selected": undefined,
                    }
                  }
                  accessibilityValue={
                    {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "space-between",
                    }
                  }
                  testID="toggle-settings-button"
                >
                  <View
                    style={
                      [
                        {
                          "padding": 10,
                        },
                        undefined,
                      ]
                    }
                    testID="wrapper"
                  >
                    <Text
                      style={
                        [
                          {
                            "color": "#808080",
                          },
                          [],
                        ]
                      }
                      testID="title"
                    >
                      Pinned Apps
                    </Text>
                  </View>
                  <Text
                    allowFontScaling={false}
                    selectable={false}
                    style={
                      [
                        {
                          "color": "#808080",
                          "fontSize": 30,
                        },
                        undefined,
                        {
                          "fontFamily": "Material Design Icons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        {},
                      ]
                    }
                  >
                    󰅀
                  </Text>
                </View>
              </View>
              <View
                style={
                  {
                    "borderRadius": 5,
                  }
                }
              >
                <View
                  accessibilityState={
                    {
                      "busy": undefined,
                      "checked": undefined,
                      "disabled": undefined,
                      "expanded": undefined,
                      "selected": undefined,
                    }
                  }
                  accessibilityValue={
                    {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "space-between",
                    }
                  }
                  testID="toggle-settings-button"
                >
                  <View
                    style={
                      [
                        {
                          "padding": 10,
                        },
                        undefined,
                      ]
                    }
                    testID="wrapper"
                  >
                    <Text
                      style={
                        [
                          {
                            "color": "#808080",
                          },
                          [],
                        ]
                      }
                      testID="title"
                    >
                      Favorite Apps
                    </Text>
                  </View>
                  <Text
                    allowFontScaling={false}
                    selectable={false}
                    style={
                      [
                        {
                          "color": "#808080",
                          "fontSize": 30,
                        },
                        undefined,
                        {
                          "fontFamily": "Material Design Icons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        {},
                      ]
                    }
                  >
                    󰅀
                  </Text>
                </View>
              </View>
              <View
                style={
                  {
                    "borderRadius": 5,
                  }
                }
              >
                <View
                  accessibilityState={
                    {
                      "busy": undefined,
                      "checked": undefined,
                      "disabled": undefined,
                      "expanded": undefined,
                      "selected": undefined,
                    }
                  }
                  accessibilityValue={
                    {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "space-between",
                    }
                  }
                  testID="toggle-settings-button"
                >
                  <View
                    style={
                      [
                        {
                          "padding": 10,
                        },
                        undefined,
                      ]
                    }
                    testID="wrapper"
                  >
                    <Text
                      style={
                        [
                          {
                            "color": "#808080",
                          },
                          [],
                        ]
                      }
                      testID="title"
                    >
                      Advanced Settings
                    </Text>
                  </View>
                  <Text
                    allowFontScaling={false}
                    selectable={false}
                    style={
                      [
                        {
                          "color": "#808080",
                          "fontSize": 30,
                        },
                        undefined,
                        {
                          "fontFamily": "Material Design Icons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        {},
                      ]
                    }
                  >
                    󰅀
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
